### PR TITLE
Adding beginning elements of score system and UI

### PR DESCRIPTION
**Pull request by dacharya64**
HUD for score should now display at the top of the screen. Does not yet update score based on player actions (that's next on my list).

### CODE REVIEW

File: assets/ui/scoreHud.ui [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+    "type": "ScoreHud",
+    "contents": {
+        "type": "relativeLayout",
+        "contents": [
+            {
+                "type": "UIText",
```
Comment (skaldarnar) [inline assets/ui/scoreHud.ui]: I guess this is a good start. For good visuals you might need to add more UI widgets here. Also, think about layout capabilities for different screen sizes, etc...

File: src/main/java/org/terasology/las/UI/ScoreHud.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    private EntityManager entityManager;
+
+    private UIText scoreArea;
+    private EntityRef scoreEntity;
+    private ScoreComponent score;
+
+    @Override
+    public void initialise() {
+        for (EntityRef entity : entityManager.getEntitiesWith(ScoreComponent.class)) {
+            scoreEntity = entity;
```
Comment (skaldarnar) [inline src/main/java/org/terasology/las/UI/ScoreHud.java]: Are you expecting one or more entities with `ScoreComponent.class`? If it is more than one, why store one of them (the last one) as `private EntityRef scoreEntity;`
Currently, the variable is only accessed in the local scope of `initialise`.

File: src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/ScoreSystem.java [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+    @Override
+    public void initialise() {
+        // Displays score UI on game start
+        String scoreScreenURI = "ScoreHud";
+        nuiManager.getHUD().addHUDElement(scoreScreenURI);
+    }
+
+    @ReceiveEvent(components = {WinConditionCheckOnActivateComponent.class})
+    public void onActivate(ActivateEvent event, EntityRef entity) {
+        LASTeam baseTeamComponent = entity.getComponent(LASTeam.class);
```
Comment (skaldarnar) [inline src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/ScoreSystem.java]: Should `LASTeam.class` be also required in the list of components here?

File: src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/ScoreSystem.java [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        EntityRef player = event.getInstigator();
+        CharacterHeldItemComponent characterHeldItemComponent = player.getComponent(CharacterHeldItemComponent.class);
+        EntityRef heldItem = characterHeldItemComponent.selectedItem;
```
Comment (skaldarnar) [inline src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/ScoreSystem.java]: Although you can be pretty sure that the player has a `CharacterHeldItemComponent.class`  component you should harden your code agains NPE by checking whether the component was actually found.
Although @vampcat and @nihal111 might know better what the best practices are here.

Comment (nihal111) [reply src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/ScoreSystem.java]: NPE checks never hurt. Would advise.

File: src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/ScoreSystem.java (hunk 3 above)
Comment (dacharya64) [reply src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/ScoreSystem.java]: The `onActivate` event seems to work with or without a LASTeam component required, does this just provide extra assurance that the base block has a team component?

File: src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/ScoreSystem.java (hunk 4 above)
Comment (dacharya64) [reply src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/ScoreSystem.java]: I might just be blanking here, but how do I check if the component was found? I know for `Entityrefs` you check if it equals `EntityRef.NULL`, but not sure for components.

File: src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/ScoreSystem.java (hunk 3 above)
Comment (nihal111) [reply src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/ScoreSystem.java]: Yes, exactly.

An event being triggered that `requires X, Y` would always trigger with just `requires X`

File: src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/ScoreSystem.java (hunk 4 above)
Comment (nihal111) [reply src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/ScoreSystem.java]: `if (characterHeldItemComponent != null)` :)

File: src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/ScoreSystem.java [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+    @In
+    private NUIManager nuiManager;
+
+    @In
+    private EntityManager entityManager;
+
+    private ControlWidget scoreScreen;
+    private EntityRef score;
+    private int redScore;
+    private int blackScore;
```
Comment (nihal111) [inline src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/ScoreSystem.java]: initialise both ints to 0

File: src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/ScoreSystem.java [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+        LASTeam baseTeamComponent = entity.getComponent(LASTeam.class);
+        EntityRef player = event.getInstigator();
+        CharacterHeldItemComponent characterHeldItemComponent = player.getComponent(CharacterHeldItemComponent.class);
+        if (characterHeldItemComponent != null) {
+            EntityRef heldItem = characterHeldItemComponent.selectedItem;
+
+            //check to see if player has other team's flag
+            if (baseTeamComponent.team.equals(baseTeamComponent.RED) && heldItem.getComponent(BlockItemComponent.class).blockFamily.getURI().toString()
+                    .equalsIgnoreCase("LightAndShadowResources:BlackFlag")) {
+                redScore++;
```
Comment (nihal111) [inline src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/ScoreSystem.java]: Reset the flag to the enemy's base before the score is incremented.

File: src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/ScoreSystem.java [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+
+    @Override
+    public void initialise() {
+        // Displays score UI on game start
+        nuiManager.getHUD().addHUDElement("ScoreHud");
+    }
+
+    @ReceiveEvent(components = {WinConditionCheckOnActivateComponent.class, LASTeam.class})
+    public void onActivate(ActivateEvent event, EntityRef entity) {
+        LASTeam baseTeamComponent = entity.getComponent(LASTeam.class);
```
Comment (nihal111) [inline src/main/java/org/terasology/ligthandshadow/componentsystem/controllers/ScoreSystem.java]: Consider renaming
LASTeam -> LASTeamComponent

File: assets/ui/scoreHud.ui [hunk 8]
```diff
@@ -0,0 +1,70 @@
+{
+    "type": "ScoreHud",
+    "contents": {
+        "type": "relativeLayout",
+        "contents": [
+                    {
```
Comment (vampcat) [inline assets/ui/scoreHud.ui]: I think I pointed this in the other review as well: indentation.

File: assets/ui/scoreHud.ui [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+                                            "use-content-width": true,
+                                            "height": 30,
+                                            "position-left": {
+                                                "widget": "redScoreLabel",
+                                                "target": "RIGHT"
+                                            }
+                                        }
+                                    }
+                                ]
+                        }
```
Comment (vampcat) [inline assets/ui/scoreHud.ui]: Indentation.

File: assets/ui/scoreHud.ui [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+                    }
+                ]
+    }
```
Comment (vampcat) [inline assets/ui/scoreHud.ui]: Indentation.

**Comment by dacharya64** [comment]
This should be good to merge now, think I covered everything in the latest push.

**Pull request closed without merging**